shared/src/types/key/ed25519.rs

```diff
@@ -448,6 +448,21 @@ impl From<PublicKey> for PublicKeyHash {
     }
 }
 
+impl From<&PublicKey> for PublicKeyHash {
```

Comment:
we're adding this in #410

Reply:
Then I will remove this when I merge in that PR.